### DISCUSSION TITLE
Where to put personal access token for Github Copilot?

### DISCUSSION THREAD

**Discussion by eliluong** [coder/code-server#6732] [Q&A] [answered]
Hello, I notice when I open VS Code in different browsers (or clear cookies), this extension will always prompt me to log into Github again. The extensions will persist between browsers, but not the Github credentials.

Is there a way put a personal access token somewhere in the configuration so that I will not have to log into Github on every machine? Thank you.

**Comment by code-asher** [comment, marked as answer]
Not at the moment.  VS Code web puts secrets in browser storage, and there does not seem to be any way to pre-load them unfortunately.  Their recommended fix seems to be to use settings sync, but that does not work with code-server from what I have been told.

Tracking: https://github.com/coder/code-server/issues/6727

**Reply by eliluong** [reply]
Thank you for letting me know the cause of the issue.

I would be okay logging in each time with a personal access token, but I have not been successful with the token auth (versus the preferred auth method).

**Reply by code-asher** [reply]
Actually, does setting the `GITHUB_TOKEN` environment variable work?  We added that a while back to authenticate the GitHub PR extension, but I was under the impression it did *not* work for Copilot.  But maybe it does?

**Reply by eliluong** [reply]
I tried to log in with GitHub PR, but even though VS Code says I am logged in, Copilot still wants me to log in. Oh, well.
